BUILD.bazel

```diff
@@ -316,3 +340,140 @@ pyx_library(
     ]),
     deps = ["//:raylet_lib"]
 )
+
+flatbuffer_py_library(
+    name = "python_gcs_fbs",
+    srcs = [
+        "src/ray/gcs/format/gcs.fbs"
+    ],
+    outs = [
+        "ActorState.py",
+        "ActorTableData.py",
+        "Arg.py",
+        "ClassTableData.py",
+        "ClientTableData.py",
+        "ConfigTableData.py",
+        "CustomSerializerData.py",
+        "DriverTableData.py",
+        "ErrorTableData.py",
+        "FunctionTableData.py",
+        "GcsTableEntry.py",
+        "HeartbeatBatchTableData.py",
+        "HeartbeatTableData.py",
+        "Language.py",
+        "ObjectTableData.py",
+        "ProfileEvent.py",
+        "ProfileTableData.py",
+        "RayResource.py",
+        "ResourcePair.py",
+        "SchedulingState.py",
+        "TablePrefix.py",
+        "TablePubsub.py",
+        "TaskInfo.py",
+        "TaskLeaseData.py",
+        "TaskReconstructionData.py",
+        "TaskTableData.py",
+        "TaskTableTestAndUpdate.py",
+    ],
+    out_prefix = "python/ray/core/generated/",
+)
+
+flatbuffer_py_library(
+    name = "python_node_manager_fbs",
+    srcs = [
+        "src/ray/raylet/format/node_manager.fbs"
+    ],
+    outs = [
+        "DisconnectClient.py",
+        "FetchOrReconstruct.py",
+        "ForwardTaskRequest.py",
+        "FreeObjectsRequest.py",
+        "GetTaskReply.py",
+        "MessageType.py",
+        "NotifyUnblocked.py",
+        "PushErrorRequest.py",
+        "RegisterClientReply.py",
+        "RegisterClientRequest.py",
+        "RegisterNodeManagerRequest.py",
+        "ResourceIdSetInfo.py",
+        "SubmitTaskRequest.py",
+        "Task.py",
+        "TaskExecutionSpecification.py",
+        "WaitReply.py",
+        "WaitRequest.py"
+    ],
+    out_prefix = "python/ray/core/generated/ray/protocol/",
+    include_paths = [
+        "src/ray/gcs/format/"
+    ],
+    includes = ["src/ray/gcs/format/gcs.fbs"]
+)
+
+filegroup(
+    name = "python_sources",
+    srcs = glob([
+        "python/ray/*.py",
+        "python/ray/autoscaler/*.py",
+        "python/ray/cloudpickle/*.py",
+        "python/ray/core/__init__.py",
+        "python/ray/core/generated/__init__.py",
+        "python/ray/core/generated/ray/__init__.py",
+        "python/ray/core/generated/ray/protocol/__init__.py",
+        "python/ray/experimental/*.py",
+        "python/ray/internal/*.py",
+        "python/ray/WebUI.ipynb",
+        "python/ray/workers/default_worker.py"
+    ])
+)
+
+genrule(
+    name = "redis",
+    cmd = "set -x && curl -sL \"https://github.com/antirez/redis/archive/5.0.3.tar.gz\" | tar xz --strip-components=1 -C . && make && find . && mv ./src/redis-server $(location redis-server)",
```

Comment:
Nice, version 5 already? I guess we may as well use the latest Redis version. Though if that causes problems we can just use the same version as before and upgrade in a separate PR.

Reply:
Yeah, they added support for streams. Let's try it and see how it goes, given what I know of redis it's probably backwards compatible but let's see when we port all the tests.